[IMP] fieldservice: Add notes on location (PM #596)

osi-scampbell opened the pull request:
This PR adds fax, email, and phone to an FSM Location record as well as implements the state_id/country_id onchange methods from res.partner.

EDIT: Added General Notes to Location as well. Ticket #607

osi-scampbell committed:
[IMP] Ticket #596 Location Field updates/additions

osi-scampbell committed:
[IMP] Flake8

max3903 commented:
@osi-scampbell There is a `partner_fax` module in OCA/partner-contact. Just add the dependency and the field on the location view.

The onchange methods should call their equivalent on `partner_id`.

osi-scampbell committed:
[IMP] Partner Fax, Tags Required

osi-scampbell commented:
> @osi-scampbell There is a `partner_fax` module in OCA/partner-contact. Just add the dependency and the field on the location view.
> 
> The onchange methods should call their equivalent on `partner_id`.

I added partner_fax and I added the comment Pat made on the ticket. I don't know how to call their equivalent on partner_id? If Odoo uses delegation inheritance then the methods don't get inherited so I am not sure how to reference them.

max3903 commented on fieldservice/models/fsm_location.py:
```diff
@@ -344,6 +344,16 @@ def _check_location_recursion(self):
+    @api.onchange('state_id')
+    def _onchange_state(self):
+        if self.state_id.country_id:
+            self.country_id = self.state_id.country_id
```
@osi-scampbell 

I was thinking of something like:
```
    @api.onchange('state_id')
    def _onchange_state(self):
        self.partner_id._onchange_state()
```

osi-scampbell replied:
self.partner_id._onchange_state() and self.partner_id._onchange_country_id() did not work for either field when I was working on these. Tested again to be doubly sure and neither method is called.

osi-scampbell committed:
[IMP] Ticket #607 Add General Notes FSM Location

max3903 commented on fieldservice/models/fsm_location.py:
```diff
@@ -65,6 +65,8 @@ class FSMLocation(models.Model):
                                default=lambda self: self._default_stage_id())
 
+    notes = fields.Text(string="Notes")
+
```
@osi-scampbell We already have a notes field on location:
https://github.com/OCA/field-service/blob/b4a901391bc7dc67a7f44c2933f221b922e39434/fieldservice/models/fsm_location.py#L48

osi-scampbell replied:
> @osi-scampbell We already have a notes field on location:
> https://github.com/OCA/field-service/blob/b4a901391bc7dc67a7f44c2933f221b922e39434/fieldservice/models/fsm_location.py#L48

I see, I don't see it on the view so I will leave the addition in the .xml but remove the duplicate in the .py

osi-scampbell committed:
[IMP] Remove duplicate field

patrickrwilson commented on fieldservice/views/fsm_location.xml:
```diff
@@ -74,6 +74,9 @@
                             <field name="owner_id"/>
                             <field name="customer_id"/>
                             <field name="contact_id" context="{'default_service_location_id': active_id, 'default_street': street, 'default_street2': street2, 'default_city': city, 'default_state_id': state_id, 'default_zip': zip, 'default_country_id': country_id}"/>
+                            <field name="fax"/>
```
add widget="phone"

patrickrwilson commented on fieldservice/views/fsm_location.xml:
```diff
@@ -74,6 +74,9 @@
                             <field name="customer_id"/>
                             <field name="contact_id" context="{'default_service_location_id': active_id, 'default_street': street, 'default_street2': street2, 'default_city': city, 'default_state_id': state_id, 'default_zip': zip, 'default_country_id': country_id}"/>
+                            <field name="fax"/>
+                            <field name="phone"/>
```
add widget="phone"

patrickrwilson commented on fieldservice/views/fsm_location.xml:
```diff
@@ -74,6 +74,9 @@
                             <field name="contact_id" context="{'default_service_location_id': active_id, 'default_street': street, 'default_street2': street2, 'default_city': city, 'default_state_id': state_id, 'default_zip': zip, 'default_country_id': country_id}"/>
+                            <field name="fax"/>
+                            <field name="phone"/>
+                            <field name="email"/>
```
add widget="email"

patrickrwilson reviewed (approved):
Just add some widgets to the phone and email fields and good to go.

osi-scampbell committed:
[IMP] phone widget

osi-scampbell committed:
[IMP] widget email

max3903 committed:
Update fsm_location.xml
